File: backend/app/schemas/food.py
from datetime import datetime
from typing import Optional, Literal
from pydantic import BaseModel, Field


class FoodItemAI(BaseModel):
    name: str
    portion: str
    calories: float
    protein_g: Optional[float] = None
    carbs_g: Optional[float] = None
    fat_g: Optional[float] = None
    fiber_g: Optional[float] = None
    confidence: float = Field(ge=0, le=1)


class AIAnalysisResponse(BaseModel):
    items: list[FoodItemAI]
    total_calories: float
    meal_notes: str = ""


class ConfirmAnalysisRequest(BaseModel):
    meal_type: Literal["breakfast", "lunch", "dinner", "snack"]
    items: list[FoodItemAI]
    photo_url: Optional[str] = None
    ai_raw_response: Optional[dict] = None


class ManualFoodEntry(BaseModel):
    meal_type: Literal["breakfast", "lunch", "dinner", "snack"]
    food_name: str = Field(min_length=1, max_length=255)
    portion_desc: Optional[str] = None
    calories: float = Field(ge=0)
    protein_g: Optional[float] = Field(default=None, ge=0)
    carbs_g: Optional[float] = Field(default=None, ge=0)
    fat_g: Optional[float] = Field(default=None, ge=0)
    fiber_g: Optional[float] = Field(default=None, ge=0)


class FoodEntryResponse(BaseModel):
    id: str
    meal_type: str
    source: str
    food_name: str
    portion_desc: Optional[str]
    calories: float
    protein_g: Optional[float]
    carbs_g: Optional[float]
    fat_g: Optional[float]
    fiber_g: Optional[float]
    photo_url: Optional[str]
    ai_confidence: Optional[float]
    is_favorite: bool
    created_at: datetime

    class Config:
        from_attributes = True


class FoodEntryUpdate(BaseModel):
    food_name: Optional[str] = None
    portion_desc: Optional[str] = None
    calories: Optional[float] = Field(default=None, ge=0)
    protein_g: Optional[float] = Field(default=None, ge=0)
    carbs_g: Optional[float] = Field(default=None, ge=0)
    fat_g: Optional[float] = Field(default=None, ge=0)
    fiber_g: Optional[float] = Field(default=None, ge=0)

File: frontend/src/types/food.ts
export type MealType = 'breakfast' | 'lunch' | 'dinner' | 'snack';
export type FoodSource = 'ai_photo' | 'ai_text' | 'manual' | 'search' | 'favorite';
export type HealthRating = 'healthy' | 'average' | 'unhealthy';

export interface FoodItemAI {
  name: string;
  portion: string;
  calories: number;
  protein_g: number | null;
  carbs_g: number | null;
  fat_g: number | null;
  fiber_g: number | null;
  confidence: number;
  health_rating?: HealthRating;
}

export interface AIAnalysisResponse {
  items: FoodItemAI[];
  total_calories: number;
  meal_notes: string;
  health_evaluation?: HealthRating;
  health_tip?: string;
}

export interface FoodEntry {
  id: string;
  meal_type: MealType;
  source: FoodSource;
  food_name: string;
  portion_desc: string | null;
  calories: number;
  protein_g: number | null;
  carbs_g: number | null;
  fat_g: number | null;
  fiber_g: number | null;
  photo_url: string | null;
  ai_confidence: number | null;
  is_favorite: boolean;
  created_at: string;
}

export interface ManualFoodEntry {
  meal_type: MealType;
  food_name: string;
  portion_desc?: string;
  calories: number;
  protein_g?: number;
  carbs_g?: number;
  fat_g?: number;
  fiber_g?: number;
}

export interface ConfirmAnalysisRequest {
  meal_type: MealType;
  items: FoodItemAI[];
  photo_url?: string;
  ai_raw_response?: Record<string, unknown>;
}
